src/datasources/human-description-api/human-description-api.service.ts

```diff
@@ -0,0 +1,124 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { IHumanDescriptionApi } from '../../domain/interfaces/human-description-api.interface';
+import {
+  MessageTemplates,
+  HumanReadableFragment,
+  ValueType,
+  Expression,
+} from './entities/human-description.entity';
+
+const isValueType = (type: unknown): type is ValueType => {
+  return Object.values(ValueType).includes(type as ValueType);
+};
+
+@Injectable()
+export class HumanDescriptionApi implements IHumanDescriptionApi {
+  private readonly parsedMessages: MessageTemplates;
+
+  /**
+   *  Regex Template that matches two patterns
+   *  1. Double curly braces consisting of 2 groups separated by a space
+   *  2. Any non-whitespace character i.e. simple words
+   */
+  private readonly TEMPLATE_REGEX = /{{(.*?)\s(\$.*?)}}|(\S+)/g;
```

Comment:
Can be `static` as I don't see this as a property of the instance itself.

Reply:
done